### PR TITLE
Fix several deprecation notices

### PR DESCRIPTION
**Pull request by XWB**
This fixes several deprecation notices:

```
Method "Twig\Extension\ExtensionInterface::getFunctions()" might add "array" as a native return type declaration in the future. Do the same in implementation "FOS\HttpCacheBundle\Twig\CacheTagExtension" now to avoid errors or add an explicit @return annotation to suppress this message.

Method "Symfony\Component\EventDispatcher\EventSubscriberInterface::getSubscribedEvents()" might add "array" as a native return type declaration in the future. Do the same in implementation "FOS\HttpCacheBundle\EventListener\CacheControlListener" now to avoid errors or add an explicit @return annotation to suppress this message.

Method "Symfony\Component\Config\Definition\ConfigurationInterface::getConfigTreeBuilder()" might add "TreeBuilder" as a native return type declaration in the future. Do the same in implementation "FOS\HttpCacheBundle\DependencyInjection\Configuration" now to avoid errors or add an explicit @return annotation to suppress this message.

Method "Symfony\Component\DependencyInjection\Extension\ConfigurationExtensionInterface::getConfiguration()" might add "?ConfigurationInterface" as a native return type declaration in the future. Do the same in implementation "FOS\HttpCacheBundle\DependencyInjection\FOSHttpCacheExtension" now to avoid errors or add an explicit @return annotation to suppress this message. 

User Deprecated: Method "Sensio\Bundle\FrameworkExtraBundle\Configuration\ConfigurationInterface::getAliasName()" might add "string" as a native return type declaration in the future. Do the same in implementation "FOS\HttpCacheBundle\Configuration\Tag" now to avoid errors or add an explicit @return annotation to suppress this message. 

User Deprecated: Method "Sensio\Bundle\FrameworkExtraBundle\Configuration\ConfigurationInterface::allowArray()" might add "bool" as a native return type declaration in the future. Do the same in implementation "FOS\HttpCacheBundle\Configuration\Tag" now to avoid errors or add an explicit @return annotation to suppress this message. 
```

### CODE REVIEW

File: src/DependencyInjection/FOSHttpCacheExtension.php [hunk 1]
```diff
@@ -37,7 +38,7 @@ class FOSHttpCacheExtension extends Extension
     /**
      * {@inheritdoc}
      */
-    public function getConfiguration(array $config, ContainerBuilder $container)
+    public function getConfiguration(array $config, ContainerBuilder $container): ?ConfigurationInterface
```
Comment (dbu) [inline src/DependencyInjection/FOSHttpCacheExtension.php:41]:
```suggestion
    public function getConfiguration(array $config, ContainerBuilder $container): ConfigurationInterface
```

i think we are allowed to be more strict about the return type than the parent class. we know we always return an instance and never null

File: src/Twig/CacheTagExtension.php [hunk 2]
```diff
@@ -55,9 +55,6 @@ public function addTag($tag)
         $this->responseTagger->addTags((array) $tag);
     }
 
-    /**
-     * {@inheritdoc}
-     */
     public function getName()
```
Comment (dbu) [inline src/Twig/CacheTagExtension.php:58]:
```suggestion
    public function getName(): string
```

**Review by dbu — COMMENTED** [review]
thanks, looks good. i added two suggestions, can you check if you agree with these?

**Comment by XWB** [comment]
@dbu Done :)

**Comment by dbu** [comment]
thank you!

**Pull request merged**